api/src/services/subscription-service.ts

```diff
@@ -63,26 +70,64 @@ export class SubscriptionService {
 		await this.channelInfoService.addChannelSubscriberId(channelAddress, subscriberId);
 		await this.channelInfoService.updateLatestChannelLink(channelAddress, res.subscriptionLink);
 
-		return res;
-	};
+		return { seed: res.seed, subscriptionLink: res.subscriptionLink };
+	}
 
-	authorizeSubscription = async (
+	async authorizeSubscription(
 		channelAddress: string,
 		subscriptionLink: string,
 		publicKey: string,
 		authorId: string
-	): Promise<{ keyloadLink: string }> => {
-		const author = await this.subscriptionPool.get(channelAddress, authorId, true);
+	): Promise<AuthorizeSubscriptionBodyResponse> {
+		const author = (await this.subscriptionPool.get(channelAddress, authorId, true)) as Author;
 		if (!author) {
 			throw new Error(`no author found with channelAddress: ${channelAddress} and identityId: ${authorId}`);
 		}
-		const authSub = await this.streamsService.authorizeSubscription(channelAddress, subscriptionLink, publicKey, <Author>author);
+		const subscriptions = await subscriptionDb.getSubscriptions(channelAddress);
+		const existingSubscriptions = subscriptions.filter((s) => s.type === SubscriptionType.Subscriber && s.isAuthorized === true);
+		const existingSubscriptionKeys = existingSubscriptions.map((s) => s.publicKey);
+
+		// fetch prev logs before syncing state
+		await this.fetchLogs(channelAddress, author, authorId);
+
+		// authorize new subscription and add existing public keys to the branch
+		await this.streamsService.receiveSubscribe(subscriptionLink, author);
+		const keyloadLink = await this.authSub(channelAddress, [publicKey, ...existingSubscriptionKeys], author, authorId, subscriptionLink);
+
+		// create new branches including the newly added subscription public key
+		await Promise.all(
```

Comment:
You can do this also with `for await` loop
```
for await (const sub of existingSubscriptions) {
    await this.authSub(channelAddress, [...existingSubscriptionKeys, publicKey], author, authorId, sub.subscriptionLink);
};

```

Reply:
Hehe ok I didn't know this, but I feel like Promise.all is more common so I would keep it for this project. But good to know thanks. :)